app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/MatchOrganisationUtrViewModel.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.eoricommoncomponent.frontend.viewModels
+
+import play.api.i18n.Messages
+import play.twirl.api.Html
+import uk.gov.hmrc.eoricommoncomponent.frontend.domain.CdsOrganisationType._
+
+object MatchOrganisationUtrViewModel {
+
+  def headerAndTitle(cdsOrgTypeString: String)(implicit messages: Messages): String = cdsOrgTypeString match {
+    case ThirdCountryOrganisationId =>
+      messages("cds.matching.row-organisation.utr.title-and-heading")
+    case orgType if individualOrganisationIds.contains(orgType) =>
+      messages("ecc.matching.row-sole-trader-individual.utr.title-and-heading")
+    case _ => messages("cds.matching.organisation.utr.title-and-heading")
+  }
+
+  def hintText(cdsOrgTypeString: String, utrLinkText: Html, utrLink: Html)(implicit messages: Messages): Html =
+    cdsOrgTypeString match {
+      case CharityPublicBodyNotForProfitId                                                  => utrLinkText
+      case ThirdCountryOrganisationId | ThirdCountryIndividualId | ThirdCountrySoleTraderId => Html(None)
+      case IndividualId | SoleTraderId =>
+        Html(messages("cds.matching.row-sole-trader-individual.utr.paragraph", utrLink))
+      case _ => Html(messages("cds.matching.organisation.utr.paragraph", utrLink))
+    }
```

Comment:
It took a while for me to work out what was going on here - can we move away from passing HTML into a messages key as a param? 

I also think it'll be worth moving the link HTML generation into this file as well rather than having some HTML in the view which only may be displayed but without having the condition in that file.

What we can do is change this object to be a singleton class and inject the paragraph HTML as well as a new link HTML, and then we can populate it with these messages and return that. We can catch up on a call